sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/RedirectPolicy.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.communication.common.implementation;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * HttpPipelinePolicy to redirect requests when 302 message is received to the new location marked by the
+ * Location header.
+ */
+public final class RedirectPolicy implements HttpPipelinePolicy {
+    private static final int MAX_REDIRECTS = 10;
+    private static final String LOCATION_HEADER_NAME = "Location";
+    private final List<String> locations = new ArrayList<>();
```

Comment:
Based on how this is being used this could be a `Set` to allow for O(1) lookup instead of O(n).

Additionally, should this set of locations be a variable passed back into the `attemptRedirection` instead of being a class-level property? A concrete example would be, could one request be redirected to a location where another request had previously been redirect before being redirected again.

Reply:
You have a valid point there, since the Policy will be there for the life of the pipeline, not for the life of the request. Thanks for the suggestion.